sky/serve/serve_utils.py

```diff
@@ -356,20 +357,32 @@ def wait_service_initialization(service_name: str, job_id: int) -> str:
     cnt = 0
     while True:
         record = serve_state.get_service_from_name(service_name)
-        if record is None:
-            continue
-        if job_id != record['controller_job_id']:
-            return common_utils.encode_payload(None)
-        lb_port = record['load_balancer_port']
-        if lb_port is not None:
-            return common_utils.encode_payload(lb_port)
+        if record is not None:
+            if job_id != record['controller_job_id']:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        f'The service {service_name!r} is already running. '
+                        'Please specify a different name for your service. '
+                        'To update an existing service, run: `sky serve down` '
+                        'and then `sky serve up` again (in-place update will '
+                        'be supported in the future).')
+            lb_port = record['load_balancer_port']
+            if lb_port is not None:
+                return common_utils.encode_payload(lb_port)
+        elif len(serve_state.get_services()) >= NUM_SERVICE_THRESHOLD:
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError('Max number of services reached. '
+                                   'To spin up more services, please '
+                                   'tear down some existing services.')
         time.sleep(1)
         cnt += 1
```

Comment:
nit: Seems we use `timeout`, why not just directly check the `time.time() - start > INITIALIZATION_TIMEOUT_SECONDS`? Seems a bit weird to compare a counter to a constant in second.